doc/superbuild/source/developers/style_guide/SourceCode.rst

```diff
@@ -338,6 +341,17 @@ Coding Conventions and Rules
    x;`` to ``return(x);``. Note, however, lots of older SUNDIALS source code
    uses ``return(x);``.
 
+#. Always use ``sunindextype`` for variables that are related to problem dimensions.
+   E.g., use it for the length of a vector, dimensions of a matrix, or for the number of vectors.
```

Comment:
I don't think we've settled on this yet so let's leave it out of the conventions until we do
```suggestion
#. Always use ``sunindextype`` for variables that are related to problem dimensions
   e.g., use it for the length of a vector or dimensions of a matrix.
```